### PR TITLE
Explicitly configure AsyncTaskExecutor

### PR DESCRIPTION
**Pull request by a1exsh**
The default `SimpleAsyncTaskExecutor` is not suitable under load (we even get a warning about it on startup):
```
An Executor is required to handle java.util.concurrent.Callable return values.
Please, configure a TaskExecutor in the MVC config under "async support".
The SimpleAsyncTaskExecutor currently in use is not suitable under load.
```

### CODE REVIEW

File: app/src/main/java/org/zalando/nakadi/config/NakadiConfig.java [hunk 1]
```diff
@@ -30,11 +28,6 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(NakadiConfig.class);
 
-    @Bean
-    public TaskExecutor taskExecutor() {
-        return new SimpleAsyncTaskExecutor();
```
Comment (adyach) [inline app/src/main/java/org/zalando/nakadi/config/NakadiConfig.java:35]: what does Spring use instead by default to manage threads?

Comment (a1exsh) [reply app/src/main/java/org/zalando/nakadi/config/NakadiConfig.java:35]: Good point — I wanted to check that, but got sidetracked :P

Looks like it uses the same one: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/context/request/async/WebAsyncManager.html#setTaskExecutor-org.springframework.core.task.AsyncTaskExecutor-

> By default a SimpleAsyncTaskExecutor instance is used.

**Comment by a1exsh** [comment]
:+1:

File: app/src/main/java/org/zalando/nakadi/config/AsyncConfig.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.concurrent.Executors;
+
+@Configuration
+public class AsyncConfig {
+    @Bean
+    @Qualifier("mvcAsyncTaskExecutor")
+    public AsyncTaskExecutor taskExecutor() {
+        return new ConcurrentTaskExecutor(Executors.newCachedThreadPool());
+    }
+}
```
Comment (antban) [inline app/src/main/java/org/zalando/nakadi/config/AsyncConfig.java:18]: @a1exsh 
Is there a reason why the configuration got to different class?

Comment (a1exsh) [reply app/src/main/java/org/zalando/nakadi/config/AsyncConfig.java:18]: Not really.  I just want to make sure that we create the thread pool only once.  If there is a way to achieve that while keeping it in the `WebConfig` class, I'd take that happily.

**Comment by a1exsh** [comment]
:+1:

**Pull request merged**